<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1 plus MathML 2.0//EN" "http://www.w3.org/Math/DTD/mathml2/xhtml-math11-f.dtd"><html xmlns="http://www.w3.org/1999/xhtml" lang="en-US"><!--This file was converted to xhtml by LibreOffice - see https://cgit.freedesktop.org/libreoffice/core/tree/filter/source/xslt for the code.--><head profile="http://dublincore.org/documents/dcmi-terms/"><meta http-equiv="Content-Type" content="application/xhtml+xml; charset=utf-8"/><title xml:lang="en-US">Cell Types/Tissues per Phenotype</title><meta name="DCTERMS.title" content="Cell Types/Tissues per Phenotype" xml:lang="en-US"/><meta name="DCTERMS.language" content="en-US" scheme="DCTERMS.RFC4646"/><meta name="DCTERMS.source" content="http://xml.openoffice.org/odf2xhtml"/><meta name="DCTERMS.issued" content="2022-02-08T12:12:35.568861652" scheme="DCTERMS.W3CDTF"/><meta name="DCTERMS.modified" content="2024-09-10T18:28:18.090026216" scheme="DCTERMS.W3CDTF"/><meta name="xsl:vendor" content="libxslt"/><link rel="schema.DC" href="http://purl.org/dc/elements/1.1/" hreflang="en"/><link rel="schema.DCTERMS" href="http://purl.org/dc/terms/" hreflang="en"/><link rel="schema.DCTYPE" href="http://purl.org/dc/dcmitype/" hreflang="en"/><link rel="schema.DCAM" href="http://purl.org/dc/dcam/" hreflang="en"/><style>
    table { border-collapse:collapse; border-spacing:0; empty-cells:show }
    td, th { vertical-align:top; font-size:10pt;}
    h1, h2, h3, h4, h5, h6 { clear:both;}
    p { white-space: nowrap; }
    ol, ul { margin:0; padding:0;}
    li { list-style: none; margin:0; padding:0;}
    span.footnodeNumber { padding-right:1em; }
    span.annotation_style_by_filter { font-size:95%; font-family:Arial; background-color:#fff000;  margin:0; border:0; padding:0;  }
    span.heading_numbering { margin-right: 0.8rem; }* { margin:0;}
    .ta1 { writing-mode:horizontal-tb; direction:ltr; }
    .Default { font-size:10pt; font-family:'Liberation Sans'; }
    .ce33 { font-size:10pt; font-family:'Liberation Sans'; border-top-width:NaNcm; border-top-style:solid; border-top-color:#000000; border-left-width:NaNcm; border-left-style:solid; border-left-color:#000000; border-bottom-width:NaNcm; border-bottom-style:solid; border-bottom-color:#000000; border-right-width:NaNcm; border-right-style:solid; border-right-color:#000000; background-color:#467599; text-align:center ! important; margin-left:0cm; color:#ffffff; font-weight:bold; }
    .ce34 { font-size:10pt; font-family:'Liberation Sans'; border-top-width:NaNcm; border-top-style:solid; border-top-color:#000000; border-left-width:NaNcm; border-left-style:solid; border-left-color:#000000; border-bottom-width:NaNcm; border-bottom-style:solid; border-bottom-color:#000000; border-right-width:NaNcm; border-right-style:solid; border-right-color:#000000; background-color:#e9fff9; vertical-align:middle; text-align:center ! important; margin-left:0cm; font-weight:bold; }
    .ce35 { font-size:10pt; font-family:'Liberation Sans'; border-top-width:NaNcm; border-top-style:solid; border-top-color:#000000; border-left-width:NaNcm; border-left-style:solid; border-left-color:#000000; border-bottom-width:NaNcm; border-bottom-style:solid; border-bottom-color:#000000; border-right-width:NaNcm; border-right-style:solid; border-right-color:#000000; background-color:#e9fff9; text-align:center ! important; margin-left:0cm; font-weight:bold; }
    .ce36 { font-size:10pt; font-family:'Liberation Sans'; border-top-width:NaNcm; border-top-style:solid; border-top-color:#000000; border-left-width:NaNcm; border-left-style:solid; border-left-color:#000000; border-bottom-width:NaNcm; border-bottom-style:solid; border-bottom-color:#000000; border-right-width:NaNcm; border-right-style:solid; border-right-color:#000000; background-color:#ffffff; vertical-align:middle; text-align:center ! important; margin-left:0cm; font-weight:bold; }
    .ce37 { font-size:10pt; font-family:'Liberation Sans'; border-top-width:NaNcm; border-top-style:solid; border-top-color:#000000; border-left-width:NaNcm; border-left-style:solid; border-left-color:#000000; border-bottom-width:NaNcm; border-bottom-style:solid; border-bottom-color:#000000; border-right-width:NaNcm; border-right-style:solid; border-right-color:#000000; background-color:#ffffff; text-align:center ! important; margin-left:0cm; font-weight:bold; }
    .ce38 { font-size:10pt; font-family:'Liberation Sans'; border-top-width:NaNcm; border-top-style:solid; border-top-color:#000000; border-left-width:NaNcm; border-left-style:solid; border-left-color:#000000; border-bottom-width:NaNcm; border-bottom-style:solid; border-bottom-color:#000000; border-right-width:NaNcm; border-right-style:solid; border-right-color:#000000; background-color:#e9fff9; vertical-align:middle; text-align:center ! important; margin-left:0cm; font-weight:bold; }
    .ce39 { font-size:10pt; font-family:'Liberation Sans'; border-top-width:NaNcm; border-top-style:solid; border-top-color:#000000; border-left-width:NaNcm; border-left-style:solid; border-left-color:#000000; border-bottom-width:NaNcm; border-bottom-style:solid; border-bottom-color:#000000; border-right-width:NaNcm; border-right-style:solid; border-right-color:#000000; background-color:#e9fff9; text-align:center ! important; margin-left:0cm; font-weight:bold; }
    .ce40 { font-size:10pt; font-family:'Liberation Sans'; }
    .ce41 { font-size:10pt; font-family:'Liberation Sans'; border-top-width:NaNcm; border-top-style:solid; border-top-color:#000000; border-left-width:NaNcm; border-left-style:solid; border-left-color:#000000; border-bottom-width:NaNcm; border-bottom-style:solid; border-bottom-color:#000000; border-right-width:NaNcm; border-right-style:solid; border-right-color:#000000; background-color:#e9fff9; text-align:center ! important; margin-left:0cm; }
    .ce42 { font-size:10pt; font-family:'Liberation Sans'; border-top-width:NaNcm; border-top-style:solid; border-top-color:#000000; border-left-width:NaNcm; border-left-style:solid; border-left-color:#000000; border-bottom-width:NaNcm; border-bottom-style:solid; border-bottom-color:#000000; border-right-width:NaNcm; border-right-style:solid; border-right-color:#000000; background-color:#ffffff; text-align:center ! important; margin-left:0cm; }
    .ce43 { font-size:10pt; font-family:'Liberation Sans'; border-top-width:NaNcm; border-top-style:solid; border-top-color:#000000; border-left-width:NaNcm; border-left-style:solid; border-left-color:#000000; border-bottom-width:NaNcm; border-bottom-style:solid; border-bottom-color:#000000; border-right-width:NaNcm; border-right-style:solid; border-right-color:#000000; background-color:#e9fff9; text-align:center ! important; margin-left:0cm; }
    .ce44 { font-size:10pt; font-family:'Liberation Sans'; border-top-width:NaNcm; border-top-style:solid; border-top-color:#000000; border-left-width:NaNcm; border-left-style:solid; border-left-color:#000000; border-bottom-width:NaNcm; border-bottom-style:solid; border-bottom-color:#000000; border-right-width:NaNcm; border-right-style:solid; border-right-color:#000000; background-color:#ffffff; text-align:center ! important; margin-left:0cm; }
    .ce45 { font-size:10pt; font-family:'Liberation Sans'; background-color:transparent; }
    .ce46 { font-size:10pt; font-family:Karumbi; }
    .ce47 { font-size:10pt; font-family:'Liberation Sans'; border-top-width:NaNcm; border-top-style:solid; border-top-color:#000000; border-left-width:NaNcm; border-left-style:solid; border-left-color:#000000; border-bottom-width:NaNcm; border-bottom-style:solid; border-bottom-color:#000000; border-right-width:NaNcm; border-right-style:solid; border-right-color:#000000; background-color:#ffffff; text-align:center ! important; margin-left:0cm; }
    .ce48 { font-size:10pt; font-family:'Liberation Sans'; background-color:transparent; }
    .ce49 { font-size:10pt; font-family:'Liberation Sans'; background-color:#afeeee; }
    .co1 { width:3.933cm; }
    .co2 { width:5.644cm; }
    .co3 { width:19.336cm; }
    .co4 { width:11.042cm; }
    .co5 { width:13.914cm; }
    .co6 { width:11.067cm; }
    .co7 { width:19.928cm; }
    .co8 { width:24.857cm; }
    .co9 { width:2.258cm; }
    .ro1 { height:0.452cm; }
    .ro2 { height:0.841cm; }
    .ro3 { height:1.221cm; }
    .ro4 { height:1.632cm; }
    .ro5 { height:1.236cm; }
    .T15 { font-family:'Liberation Sans'; font-size:10pt; font-weight:normal; text-decoration:none ! important; font-style:normal; text-shadow:none; }
    .T16 { font-size:10pt; font-weight:normal; text-decoration:none ! important; font-style:normal; text-shadow:none; }
    .T17 { font-family:'Liberation Sans'; font-weight:normal; text-decoration:none ! important; font-style:normal; text-shadow:none; }
    .T18 { font-family:'Liberation Sans'; text-shadow:none; font-style:normal; text-decoration:none ! important; font-weight:normal; font-size:10pt; }
    .T19 { font-family:'Liberation Sans'; text-shadow:none; font-style:normal; text-decoration:none ! important; font-weight:normal; }
    .T20 { font-family:'Liberation Sans'; }
    .T21 { font-family:'Liberation Sans'; font-weight:normal; text-decoration:none ! important; font-style:normal; text-shadow:none; }
    .T22 { font-family:'Liberation Sans'; font-size:10pt; font-weight:normal; text-decoration:none ! important; font-style:normal; text-shadow:none; }
    .T23 { font-family:'Liberation Sans'; font-size:10pt; font-weight:normal; text-decoration:none ! important; font-style:normal; text-shadow:none; }
    /* ODF styles with no properties representable as CSS:
     { } */
</style></head><body dir="ltr"><table border="0" cellspacing="0" cellpadding="0" class="ta1"><colgroup><col width="172"/><col width="247"/><col width="845"/><col width="483"/><col width="608"/><col width="484"/><col width="871"/><col width="871"/><col width="1086"/><col width="99"/><col width="99"/></colgroup><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="ce33"><p>Phenotype</p></td><td style="text-align:left;width:5.644cm; " class="ce33"><p>  Cell Types/Tissues  </p></td><td style="text-align:left;width:19.336cm; " class="ce33"><p> Cell Types/Tissues additional information </p></td><td style="text-align:left;width:11.042cm; " class="ce33"><p>Chip-Seq Source for enhancer annotation</p></td><td style="text-align:left;width:13.914cm; " class="ce33"><p>Enhancer annotation (based on Chip-Seq), Main Procedure</p></td><td style="text-align:left;width:11.067cm; " class="ce33"><p>RNA-Seq Source</p></td><td style="text-align:left;width:19.928cm; " class="ce33"><p>Gene Expression quantification main procedure</p></td><td style="text-align:left;width:19.928cm; " class="ce33"><p>HiC – TAD maps Source </p></td><td style="text-align:left;width:24.857cm; " class="ce33"><p>TAD maps main procedure</p></td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro2"><td rowspan="4" style="text-align:left;width:3.933cm; " class="ce34"><p>Cardiovascular</p></td><td style="text-align:left;width:5.644cm; " class="ce41"><p>Day 5 </p></td><td style="text-align:left;width:19.336cm; " class="ce41"><p>Day 5 cardiac mesodermal cells, from 80 days ESC differentiation towards ventricular cardiomyocytes</p><p>For + info: <a href="https://www.nature.com/articles/s41588-019-0479-7">https://www.nature.com/articles/s41588-019-0479-7</a> </p></td><td style="text-align:left;width:11.042cm; " class="ce41"><p><span class="T17"><a href="https://doi.org/10.1038/s41588-019-0479-7">https://doi.org/10.1038/s41588-019-0479-7</a></span></p></td><td style="text-align:left;width:13.914cm; " class="ce41"><p>H3K27ac peak with distance &gt; 10Kb to protein coding TSS</p><p>(Peaks were called after reads mapping with Bowtie2 with Macs2)</p></td><td style="text-align:left;width:11.067cm; " class="ce41"><p><span class="T17"><a href="https://doi.org/10.1038/s41588-019-0479-7">https://doi.org/10.1038/s41588-019-0479-7</a></span></p></td><td style="text-align:left;width:19.928cm; " class="ce41"><p>FPKMs directly obtained from source</p></td><td style="text-align:left;width:19.928cm; " class="ce41"><p><span class="T17"> </span><span class="T17"><a href="https://doi.org/10.1038/s41588-019-0479-7">https://doi.org/10.1038/s41588-019-0479-7</a></span></p></td><td style="text-align:left;width:24.857cm; " class="ce41"><p>TAD map generated upon processing Day 5 cardiomyocyte differentiation HiC data.</p><p> TADs were called with the usage of the tool  DomainCaller by considering the 50kb contact matrices in the .hic files</p></td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro2"><td style="text-align:left;width:5.644cm; " class="ce41"><p>Day 7</p></td><td style="text-align:left;width:19.336cm; " class="ce41"><p>Day 7 cardiac progenitors, from 80 days ESC differentiation towards ventricular cardiomyocytes</p><p><span class="T15">For + info: </span><span class="T15"><a href="https://www.nature.com/articles/s41588-019-0479-7">https://www.nature.com/articles/s41588-019-0479-7</a></span><span class="T15"> </span></p></td><td style="text-align:left;width:11.042cm; " class="ce41"><p><span class="T17"><a href="https://doi.org/10.1038/s41588-019-0479-7">https://doi.org/10.1038/s41588-019-0479-7</a></span></p></td><td style="text-align:left;width:13.914cm; " class="ce41"><p><span class="T15">H3K27ac peak with distance &gt; 10Kb to protein coding TSS</span></p><p><span class="T15">(Peaks were called after reads mapping with Bowtie2 with Macs2)</span></p></td><td style="text-align:left;width:11.067cm; " class="ce41"><p><span class="T17"><a href="https://doi.org/10.1038/s41588-019-0479-7">https://doi.org/10.1038/s41588-019-0479-7</a></span></p></td><td style="text-align:left;width:19.928cm; " class="ce41"><p>FPKMs directly obtained from source</p></td><td style="text-align:left;width:19.928cm; " class="ce41"><p><span class="T17"> </span><span class="T17"><a href="https://doi.org/10.1038/s41588-019-0479-7">https://doi.org/10.1038/s41588-019-0479-7</a></span></p></td><td style="text-align:left;width:24.857cm; " class="ce41"><p>TAD map generated upon processing Day 7 cardiomyocyte differentiation HiC data.</p><p> TADs were called with the usage of the tool  DomainCaller by considering the 50kb contact matrices in the .hic files</p></td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro2"><td style="text-align:left;width:5.644cm; " class="ce41"><p>Day 15</p></td><td style="text-align:left;width:19.336cm; " class="ce41"><p>Day 15 primitive cardiomyocytes, from 80 days ESC differentiation towards ventricular cardiomyocytes</p><p><span class="T15">For + info: </span><span class="T15"><a href="https://www.nature.com/articles/s41588-019-0479-7">https://www.nature.com/articles/s41588-019-0479-7</a></span><span class="T15"> </span></p></td><td style="text-align:left;width:11.042cm; " class="ce41"><p><span class="T17"><a href="https://doi.org/10.1038/s41588-019-0479-7">https://doi.org/10.1038/s41588-019-0479-7</a></span></p></td><td style="text-align:left;width:13.914cm; " class="ce41"><p><span class="T15">H3K27ac peak with distance &gt; 10Kb to protein coding TSS</span></p><p><span class="T15">(Peaks were called after reads mapping with Bowtie2 with Macs2)</span></p></td><td style="text-align:left;width:11.067cm; " class="ce41"><p><span class="T17"><a href="https://doi.org/10.1038/s41588-019-0479-7">https://doi.org/10.1038/s41588-019-0479-7</a></span></p></td><td style="text-align:left;width:19.928cm; " class="ce41"><p>FPKMs directly obtained from source</p></td><td style="text-align:left;width:19.928cm; " class="ce41"><p><span class="T17"> </span><span class="T17"><a href="https://doi.org/10.1038/s41588-019-0479-7">https://doi.org/10.1038/s41588-019-0479-7</a></span></p></td><td style="text-align:left;width:24.857cm; " class="ce41"><p>TAD map generated upon processing Day 15 cardiomyocyte differentiation HiC data.</p><p>TADs were called with the usage of the tool  DomainCaller by considering the 50kb contact matrices in the .hic files</p></td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro2"><td style="text-align:left;width:5.644cm; " class="ce41"><p>Day 80</p></td><td style="text-align:left;width:19.336cm; " class="ce41"><p>Day 80 ventricular cardiomiocytes, from 80 days ESC differentiation towards ventricular cardiomyocytes</p><p><span class="T15">For + info: </span><span class="T15"><a href="https://www.nature.com/articles/s41588-019-0479-7">https://www.nature.com/articles/s41588-019-0479-7</a></span><span class="T15"> </span></p></td><td style="text-align:left;width:11.042cm; " class="ce41"><p><span class="T17"><a href="https://doi.org/10.1038/s41588-019-0479-7">https://doi.org/10.1038/s41588-019-0479-7</a></span></p></td><td style="text-align:left;width:13.914cm; " class="ce41"><p>H3K27ac peak with distance &gt; 10Kb to protein coding TSS</p><p>(Peaks were called after reads mapping with Bowtie2 with Macs2)</p></td><td style="text-align:left;width:11.067cm; " class="ce41"><p><span class="T17"><a href="https://doi.org/10.1038/s41588-019-0479-7">https://doi.org/10.1038/s41588-019-0479-7</a></span></p></td><td style="text-align:left;width:19.928cm; " class="ce41"><p>FPKMs directly obtained from source</p></td><td style="text-align:left;width:19.928cm; " class="ce41"><p><span class="T17"> </span><span class="T17"><a href="https://doi.org/10.1038/s41588-019-0479-7">https://doi.org/10.1038/s41588-019-0479-7</a></span></p></td><td style="text-align:left;width:24.857cm; " class="ce41"><p>TAD map generated upon processing Day 80 cardiomyocyte differentiation HiC data.</p><p>TADs were called with the usage of the tool  DomainCaller by considering the 50kb contact matrices in the .hic files</p></td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro3"><td rowspan="3" style="text-align:left;width:3.933cm; " class="ce36"><p>Head-Neck</p></td><td style="text-align:left;width:5.644cm; " class="ce42"><p>Neural Crest Early</p></td><td style="text-align:left;width:19.336cm; " class="ce42"><p>Neural Crest Data  of hNCC differentiation</p><p>This NC data corresponds with an early and heterogeneous stage of NC differentiation</p><p><span class="T15">For +info: </span><span class="T15"><a href="https://pubmed.ncbi.nlm.nih.gov/22981823/">https://pubmed.ncbi.nlm.nih.gov/22981823/</a></span></p></td><td style="text-align:left;width:11.042cm; " class="ce42"><p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE28876">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE28876</a></p></td><td style="text-align:left;width:13.914cm; " class="ce42"><p>p300 peak, intersecting H3K27ac peak, with distance &gt; 10Kb to protein coding TSS</p><p><span class="T15">(Peaks were called upon reads mapping with Bowtie2 with Macs2)</span></p></td><td style="text-align:left;width:11.067cm; " class="ce42"><p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE28876">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE28876</a></p></td><td style="text-align:left;width:19.928cm; " class="ce42"><p>FPKMs obtained upon reads mapping with Tophat2 and gene expression quantification with Cufflinks</p></td><td style="text-align:left;width:19.928cm; " class="ce42"><p><span class="T17">3D genome browser: </span><span class="T17"><a href="http://3dgenome.fsm.northwestern.edu/">http://3dgenome.fsm.northwestern.edu/</a></span></p><p><span class="T17">TAD maps: </span><span class="T17"><a href="http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip">http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip</a></span></p></td><td style="text-align:left;width:24.857cm; " class="ce42"><p><span class="T15">ESC TAD map directly obtained from: </span><span class="T15"><a href="http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip">http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip</a></span></p></td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro3"><td style="text-align:left;width:5.644cm; " class="ce42"><p>Neural Crest Late</p></td><td style="text-align:left;width:19.336cm; " class="ce44"><p><span class="T16">Neural Crest Data  of hNCC differentiation </span></p><p><span class="T16">P4 differentiation stage, it corresponds with a later and more homogeneous stage of NC differentation in comparison with Neural Crest Early</span></p><p><span class="T16">For +info: </span><span class="T16"><a href="https://pubmed.ncbi.nlm.nih.gov/26365491/">https://pubmed.ncbi.nlm.nih.gov/26365491/</a></span></p></td><td style="text-align:left;width:11.042cm; " class="ce42"><p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE70751">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE70751</a></p></td><td style="text-align:left;width:13.914cm; " class="ce42"><p><span class="T15">p300 peak, intersecting H3K27ac peak, with distance &gt; 10Kb to protein coding TSS</span></p><p><span class="T15">(Peaks were called upon reads mapping with Bowtie2 with Macs2)</span></p></td><td style="text-align:left;width:11.067cm; " class="ce42"><p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE70751">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE70751</a></p></td><td style="text-align:left;width:19.928cm; " class="ce42"><p>FPKMs obtained upon reads mapping with Tophat2 and  gene expression quantification with Cufflinks</p></td><td style="text-align:left;width:19.928cm; " class="ce42"><p><span class="T15">3D genome browser: </span><span class="T15"><a href="http://3dgenome.fsm.northwestern.edu/">http://3dgenome.fsm.northwestern.edu/</a></span></p><p><span class="T15">TAD maps: </span><span class="T15"><a href="http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip">http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip</a></span></p></td><td style="text-align:left;width:24.857cm; " class="ce42"><p><span class="T15">ESC TAD map directly obtained from: </span><span class="T15"><a href="http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip">http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip</a></span></p></td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro2"><td style="text-align:left;width:5.644cm; " class="ce42"><p>PalateCS20 </p></td><td style="text-align:left;width:19.336cm; " class="ce42"><p>Embryonic Palate Carnagie Stage 20</p><p><span class="T15">For +info: </span><span class="T15"><a href="https://pubmed.ncbi.nlm.nih.gov/29719267/">https://pubmed.ncbi.nlm.nih.gov/29719267/</a></span></p></td><td style="text-align:left;width:11.042cm; " class="ce42"><p><a href="https://pubmed.ncbi.nlm.nih.gov/29719267/">https://pubmed.ncbi.nlm.nih.gov/29719267/</a></p></td><td style="text-align:left;width:13.914cm; " class="ce42"><p>H3K27ac peak with distance &gt; 10Kb to protein coding TSS</p><p>(Peaks were called with Macs2 using the provided alignment files (tagAlign))</p></td><td style="text-align:left;width:11.067cm; " class="ce42"><p><a href="https://elifesciences.org/articles/15657">https://elifesciences.org/articles/15657</a></p></td><td style="text-align:left;width:19.928cm; " class="ce42"><p>To obtain the FPKMs, first, the bedfiles containing the reads mapping coordinates (.unique.bed12) for Palate1 sample were converted into bam files with bedToBam tool.</p><p> Next FPKMs were computed with Cufflinks.</p></td><td style="text-align:left;width:19.928cm; " class="ce42"><p><span class="T15">3D genome browser: </span><span class="T15"><a href="http://3dgenome.fsm.northwestern.edu/">http://3dgenome.fsm.northwestern.edu/</a></span></p><p><span class="T15">TAD maps: </span><span class="T15"><a href="http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip">http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip</a></span></p></td><td style="text-align:left;width:24.857cm; " class="ce42"><p><span class="T15">ESC TAD map directly obtained from: </span><span class="T15"><a href="http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip">http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip</a></span></p></td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro2"><td rowspan="2" style="text-align:left;width:3.933cm; " class="ce34"><p>Limbs</p></td><td style="text-align:left;width:5.644cm; " class="ce41"><p>EmbryonicLimb1</p></td><td style="text-align:left;width:19.336cm; " class="ce41"><p>Embryonic Lower Limb data of entire limb buds, Carnegie Stages 14-19</p><p><span class="T15">For +info: </span><span class="T15"><a href="https://www.nature.com/articles/s41467-020-17305-2">https://www.nature.com/articles/s41467-020-17305-2</a></span><span class="T15"> </span></p></td><td style="text-align:left;width:11.042cm; " class="ce41"><p><a href="https://www.nature.com/articles/s41467-020-17305-2">https://www.nature.com/articles/s41467-020-17305-2</a></p></td><td style="text-align:left;width:13.914cm; " class="ce41"><p>H3K27ac peak with distance &gt; 10Kb to protein coding TSS</p><p><span class="T15">(Peaks were called after reads mapping with Bowtie2 with Macs2)</span></p></td><td style="text-align:left;width:11.067cm; " class="ce41"><p><a href="https://elifesciences.org/articles/15657">https://elifesciences.org/articles/15657</a></p><p> </p></td><td style="text-align:left;width:19.928cm; " class="ce41"><p><span class="T15">To obtain the FPKMs, first, the bedfiles containing the reads mapping coordinates (.unique.bed12) for Lower Limb samples were converted into bam files with bedToBam tool.</span></p><p><span class="T15"> Next FPKMs were computed with Cufflinks.</span></p></td><td style="text-align:left;width:19.928cm; " class="ce41"><p><span class="T15">3D genome browser: </span><span class="T15"><a href="http://3dgenome.fsm.northwestern.edu/">http://3dgenome.fsm.northwestern.edu/</a></span></p><p><span class="T15">TAD maps: </span><span class="T15"><a href="http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip">http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip</a></span></p></td><td style="text-align:left;width:24.857cm; " class="ce41"><p><span class="T15">ESC TAD map directly obtained from: </span><span class="T15"><a href="http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip">http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip</a></span></p></td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro2"><td style="text-align:left;width:5.644cm; " class="ce41"><p>EmbryonicLimb2</p></td><td style="text-align:left;width:19.336cm; " class="ce41"><p>Embryonic Upper Limb data <span class="T15">of entire limb buds, Carnegie Stages 14-19</span></p><p><span class="T15">For +info: </span><span class="T15"><a href="https://www.nature.com/articles/s41467-020-17305-2">https://www.nature.com/articles/s41467-020-17305-2</a></span><span class="T15"> </span></p></td><td style="text-align:left;width:11.042cm; " class="ce41"><p><a href="https://www.nature.com/articles/s41467-020-17305-2">https://www.nature.com/articles/s41467-020-17305-2</a></p></td><td style="text-align:left;width:13.914cm; " class="ce41"><p>H3K27ac peak with distance &gt; 10Kb to protein coding TSS</p><p><span class="T15">(Peaks were called after reads mapping with Bowtie2 with Macs2)</span></p></td><td style="text-align:left;width:11.067cm; " class="ce41"><p><a href="https://elifesciences.org/articles/15657">https://elifesciences.org/articles/15657</a></p><p> </p></td><td style="text-align:left;width:19.928cm; " class="ce41"><p><span class="T15">To obtain the FPKMs, first, the bedfiles containing the reads mapping coordinates (.unique.bed12) for Upper Limb samples were converted into bam files with bedToBam tool.</span></p><p><span class="T15"> Next FPKMs were computed with Cufflinks.</span></p></td><td style="text-align:left;width:19.928cm; " class="ce41"><p><span class="T15">3D genome browser: </span><span class="T15"><a href="http://3dgenome.fsm.northwestern.edu/">http://3dgenome.fsm.northwestern.edu/</a></span></p><p><span class="T15">TAD maps: </span><span class="T15"><a href="http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip">http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip</a></span></p></td><td style="text-align:left;width:24.857cm; " class="ce41"><p><span class="T15">ESC TAD map directly obtained from: </span><span class="T15"><a href="http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip">http://3dgenome.fsm.northwestern.edu/downloads/hg19.TADs.zip</a></span></p></td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro4"><td rowspan="2" style="text-align:left;width:3.933cm; " class="ce36"><p>Neurodevelopmental</p></td><td style="text-align:left;width:5.644cm; " class="ce42"><p>PfcGw15</p></td><td style="text-align:left;width:19.336cm; " class="ce42"><p>Brain prefrontal cortex Gestation Week 15 </p><p>For +info: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149268">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149268</a> </p></td><td style="text-align:left;width:11.042cm; " class="ce42"><p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149268">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149268</a></p></td><td style="text-align:left;width:13.914cm; " class="ce42"><p>H3K27ac peak with distance &gt; 10Kb to protein coding TSS</p><p><span class="T15">(Peaks were directly obtained from source)</span></p></td><td style="text-align:left;width:11.067cm; " class="ce42"><p><span class="T20">From database: </span><span class="T20"><a href="https://www.brainspan.org/static/home">https://www.brainspan.org/static/home</a></span></p><p> </p></td><td style="text-align:left;width:19.928cm; " class="ce42"><p><span class="T22">Expression values obtained from file: summarized to genes file</span></p><p><span class="T22">Given that gestational age is 2 weeks longer than conceptional age,  post conception week 13 (pcw13) samples information was considered to match with chip-seq gestation week 15 data</span></p><p><span class="T23">Expression value computed as the average of the pcw13 samples: VFC (ventrolateral prefrontal cortex), MFC[anterior (rostral) cingulate (medial prefrontal) cortex] and DFC(dorsolateral prefrontal cortex)</span></p><p> </p></td><td style="text-align:left;width:19.928cm; " class="ce42"><p><a href="https://doi.org/10.1016/j.celrep.2016.10.061">https://doi.org/10.1016/j.celrep.2016.10.061</a></p></td><td style="text-align:left;width:24.857cm; " class="ce42"><p>Brain Prefrontal Cortex TAD map, generated from CO (prefrontal cortex) boundary map provided in the referenced source.</p><p>The Excelfile containing the boundary maps can be found here:</p><p> <a href="https://www.cell.com/cms/10.1016/j.celrep.2016.10.061/attachment/c3954472-3378-4fd7-aa33-9a2d085f7bbd/mmc4.xlsx">https://www.cell.com/cms/10.1016/j.celrep.2016.10.061/attachment/c3954472-3378-4fd7-aa33-9a2d085f7bbd/mmc4.xlsx</a></p></td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro4"><td style="text-align:left;width:5.644cm; " class="ce42"><p>PfcGw18</p></td><td style="text-align:left;width:19.336cm; " class="ce42"><p>Brain prefrontal cortex Gestation Week 18</p><p>For +info: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149268">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149268</a> </p></td><td style="text-align:left;width:11.042cm; " class="ce42"><p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149268">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149268</a></p></td><td style="text-align:left;width:13.914cm; " class="ce42"><p>H3K27ac peak with distance &gt; 10Kb to protein coding TSS</p><p>(Peaks were directly obtained from source)</p></td><td style="text-align:left;width:11.067cm; " class="ce42"><p><span class="T21">From database: </span><span class="T21"><a href="https://www.brainspan.org/static/home">https://www.brainspan.org/static/home</a></span></p><p> </p></td><td style="text-align:left;width:19.928cm; " class="ce47"><p><span class="T22">Expression values obtained from file: summarized to genes file</span></p><p><span class="T22">Given that gestational age is 2 weeks longer than conceptional age,  post conception week 16 (pcw16) samples information was considered to match with chip-seq gestation week 18 data</span></p><p><span class="T22">Expression value computed as the average of the pcw16 samples: VFC (ventrolateral prefrontal cortex), MFC[anterior (rostral) cingulate (medial prefrontal) cortex] and DFC(dorsolateral prefrontal cortex)</span></p><p> </p></td><td style="text-align:left;width:19.928cm; " class="ce42"><p><span class="T17"><a href="https://doi.org/10.1016/j.celrep.2016.10.061">https://doi.org/10.1016/j.celrep.2016.10.061</a></span></p></td><td style="text-align:left;width:24.857cm; " class="ce42"><p>Brain Prefrontal Cortex TAD map, generated from CO (prefrontal cortex) boundary map provided in the referenced source.</p><p>The Excelfile containing the boundary maps can be found here:</p><p> <a href="https://www.cell.com/cms/10.1016/j.celrep.2016.10.061/attachment/c3954472-3378-4fd7-aa33-9a2d085f7bbd/mmc4.xlsx">https://www.cell.com/cms/10.1016/j.celrep.2016.10.061/attachment/c3954472-3378-4fd7-aa33-9a2d085f7bbd/mmc4.xlsx</a></p></td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro5"><td rowspan="2" style="text-align:left;width:3.933cm; " class="ce38"><p>Vision-Eye</p></td><td style="text-align:left;width:5.644cm; " class="ce43"><p>Retina</p></td><td style="text-align:left;width:19.336cm; " class="ce43"><p>Retina (from Adults)</p><p>For +info: <a href="https://pubmed.ncbi.nlm.nih.gov/32265282/">https://pubmed.ncbi.nlm.nih.gov/32265282/</a></p></td><td style="text-align:left;width:11.042cm; " class="ce43"><p><span class="T18"><a href="https://pubmed.ncbi.nlm.nih.gov/32265282/">https://pubmed.ncbi.nlm.nih.gov/32265282/</a></span><span class="T18"> </span></p></td><td style="text-align:left;width:13.914cm; " class="ce43"><p><span class="T15">Liftover of enhancer coordinates from hg38 to hg19</span>, as provided by source</p><p>(derived of ATAC peaks overlapping H3K27ac peaks) .</p><p>In addition enhancers &lt; 2.5kb from any protein coding TSS were discarded.</p></td><td style="text-align:left;width:11.067cm; " class="ce43"><p><span class="T19"><a href="https://pubmed.ncbi.nlm.nih.gov/32265282/">https://pubmed.ncbi.nlm.nih.gov/32265282/</a></span><span class="T19"> </span></p></td><td style="text-align:left;width:19.928cm; " class="ce43"><p>FPKMs obtained upon reads mapping with Hisat2 and gene expression quantification with Cufflinks</p></td><td style="text-align:left;width:19.928cm; " class="ce43"><p><a href="https://pubmed.ncbi.nlm.nih.gov/36207300/">https://pubmed.ncbi.nlm.nih.gov/36207300/</a></p></td><td style="text-align:left;width:24.857cm; " class="ce43"><p>Retina TAD map obtained from source after liftovering coordinates from hg38 to hg19. </p><p>Since no TADs for chrX were provided in the source, those from ESCs were considered as reference for chrX.</p></td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro5"><td style="text-align:left;width:5.644cm; " class="ce43"><p>RPE</p></td><td style="text-align:left;width:19.336cm; " class="ce43"><p>Retina Pigment Epithelium (from Adults)</p><p>For +info: <a href="https://pubmed.ncbi.nlm.nih.gov/32265282/">https://pubmed.ncbi.nlm.nih.gov/32265282/</a> </p></td><td style="text-align:left;width:11.042cm; " class="ce43"><p><span class="T19"><a href="https://pubmed.ncbi.nlm.nih.gov/32265282/">https://pubmed.ncbi.nlm.nih.gov/32265282/</a></span><span class="T19"> </span></p></td><td style="text-align:left;width:13.914cm; " class="ce43"><p><span class="T15">Liftover of enhancer coordinates from hg38 to hg19, as provided by source</span></p><p><span class="T15">(derived of Dnase-seq peaks overlapping H3K27ac peaks) .</span></p><p><span class="T15">In addition enhancers &lt; 2.5kb from any protein coding TSS were discarded.</span></p></td><td style="text-align:left;width:11.067cm; " class="ce43"><p><span class="T19"><a href="https://pubmed.ncbi.nlm.nih.gov/32265282/">https://pubmed.ncbi.nlm.nih.gov/32265282/</a></span><span class="T19"> </span></p></td><td style="text-align:left;width:19.928cm; " class="ce43"><p>FPKMs obtained upon reads mapping with Hisat2 and gene expression quantification with Cufflinks</p></td><td style="text-align:left;width:19.928cm; " class="ce43"><p><a href="https://pubmed.ncbi.nlm.nih.gov/36207300/">https://pubmed.ncbi.nlm.nih.gov/36207300/</a></p></td><td style="text-align:left;width:24.857cm; " class="ce43"><p>Retina TAD map obtained from source after liftovering coordinates from hg38 to hg19.</p><p><span class="T15">Since no TADs for chrX were provided in the source, those from ESCs were considered as reference for chrX.</span></p></td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="ce40"> </td><td style="text-align:left;width:5.644cm; " class="ce40"> </td><td style="text-align:left;width:19.336cm; " class="ce40"> </td><td style="text-align:left;width:11.042cm; " class="ce40"> </td><td style="text-align:left;width:13.914cm; " class="ce40"> </td><td style="text-align:left;width:11.067cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:24.857cm; " class="ce40"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="ce40"> </td><td style="text-align:left;width:5.644cm; " class="ce40"> </td><td style="text-align:left;width:19.336cm; " class="ce40"> </td><td style="text-align:left;width:11.042cm; " class="ce40"> </td><td style="text-align:left;width:13.914cm; " class="ce40"> </td><td style="text-align:left;width:11.067cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:24.857cm; " class="ce40"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="ce40"> </td><td style="text-align:left;width:5.644cm; " class="ce40"> </td><td style="text-align:left;width:19.336cm; " class="ce40"> </td><td style="text-align:left;width:11.042cm; " class="ce40"> </td><td style="text-align:left;width:13.914cm; " class="ce40"> </td><td style="text-align:left;width:11.067cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:24.857cm; " class="ce40"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="ce40"> </td><td style="text-align:left;width:5.644cm; " class="ce40"> </td><td style="text-align:left;width:19.336cm; " class="ce40"> </td><td style="text-align:left;width:11.042cm; " class="ce40"> </td><td style="text-align:left;width:13.914cm; " class="ce40"> </td><td style="text-align:left;width:11.067cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:24.857cm; " class="ce40"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="ce40"> </td><td style="text-align:left;width:5.644cm; " class="ce40"> </td><td style="text-align:left;width:19.336cm; " class="ce40"> </td><td style="text-align:left;width:11.042cm; " class="ce40"> </td><td style="text-align:left;width:13.914cm; " class="ce40"> </td><td style="text-align:left;width:11.067cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:24.857cm; " class="ce40"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="ce40"> </td><td style="text-align:left;width:5.644cm; " class="ce40"> </td><td style="text-align:left;width:19.336cm; " class="ce40"> </td><td style="text-align:left;width:11.042cm; " class="ce40"> </td><td style="text-align:left;width:13.914cm; " class="ce40"> </td><td style="text-align:left;width:11.067cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:24.857cm; " class="ce40"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="ce40"> </td><td style="text-align:left;width:5.644cm; " class="ce40"> </td><td style="text-align:left;width:19.336cm; " class="ce40"> </td><td style="text-align:left;width:11.042cm; " class="ce40"> </td><td style="text-align:left;width:13.914cm; " class="ce40"> </td><td style="text-align:left;width:11.067cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:24.857cm; " class="ce40"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="ce40"> </td><td style="text-align:left;width:5.644cm; " class="ce40"> </td><td style="text-align:left;width:19.336cm; " class="ce45"> </td><td style="text-align:left;width:11.042cm; " class="ce40"> </td><td style="text-align:left;width:13.914cm; " class="ce40"> </td><td style="text-align:left;width:11.067cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:24.857cm; " class="ce40"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="ce40"> </td><td style="text-align:left;width:5.644cm; " class="ce40"> </td><td style="text-align:left;width:19.336cm; " class="ce40"> </td><td style="text-align:left;width:11.042cm; " class="ce40"> </td><td style="text-align:left;width:13.914cm; " class="ce40"> </td><td style="text-align:left;width:11.067cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:24.857cm; " class="ce40"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="ce40"> </td><td style="text-align:left;width:5.644cm; " class="ce40"> </td><td style="text-align:left;width:19.336cm; " class="ce40"> </td><td style="text-align:left;width:11.042cm; " class="ce40"> </td><td style="text-align:left;width:13.914cm; " class="ce40"> </td><td style="text-align:left;width:11.067cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:24.857cm; " class="ce40"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="ce40"> </td><td style="text-align:left;width:5.644cm; " class="ce40"> </td><td style="text-align:left;width:19.336cm; " class="ce40"> </td><td style="text-align:left;width:11.042cm; " class="ce40"> </td><td style="text-align:left;width:13.914cm; " class="ce40"> </td><td style="text-align:left;width:11.067cm; " class="ce46"> </td><td style="text-align:left;width:19.928cm; " class="ce46"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:24.857cm; " class="ce40"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="ce40"> </td><td style="text-align:left;width:5.644cm; " class="ce40"> </td><td style="text-align:left;width:19.336cm; " class="ce40"> </td><td style="text-align:left;width:11.042cm; " class="ce40"> </td><td style="text-align:left;width:13.914cm; " class="ce40"> </td><td style="text-align:left;width:11.067cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:24.857cm; " class="ce40"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="ce40"> </td><td style="text-align:left;width:5.644cm; " class="ce40"> </td><td style="text-align:left;width:19.336cm; " class="ce40"> </td><td style="text-align:left;width:11.042cm; " class="ce40"> </td><td style="text-align:left;width:13.914cm; " class="ce40"> </td><td style="text-align:left;width:11.067cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:24.857cm; " class="ce40"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="ce40"> </td><td style="text-align:left;width:5.644cm; " class="ce40"> </td><td style="text-align:left;width:19.336cm; " class="ce40"> </td><td style="text-align:left;width:11.042cm; " class="ce40"> </td><td style="text-align:left;width:13.914cm; " class="ce40"> </td><td style="text-align:left;width:11.067cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:24.857cm; " class="ce40"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="ce40"> </td><td style="text-align:left;width:5.644cm; " class="ce40"> </td><td style="text-align:left;width:19.336cm; " class="ce40"> </td><td style="text-align:left;width:11.042cm; " class="ce40"> </td><td style="text-align:left;width:13.914cm; " class="ce40"> </td><td style="text-align:left;width:11.067cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:19.928cm; " class="ce40"> </td><td style="text-align:left;width:24.857cm; " class="ce40"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td><td style="text-align:left;width:2.258cm; " class="ce45"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="Default"> </td><td style="text-align:left;width:5.644cm; " class="Default"> </td><td style="text-align:left;width:19.336cm; " class="Default"> </td><td style="text-align:left;width:11.042cm; " class="Default"> </td><td style="text-align:left;width:13.914cm; " class="Default"> </td><td style="text-align:left;width:11.067cm; " class="Default"> </td><td style="text-align:left;width:19.928cm; " class="Default"> </td><td style="text-align:left;width:19.928cm; " class="Default"> </td><td style="text-align:left;width:24.857cm; " class="Default"> </td><td style="text-align:left;width:2.258cm; " class="ce48"> </td><td style="text-align:left;width:2.258cm; " class="Default"> </td></tr><tr class="ro1"><td style="text-align:left;width:3.933cm; " class="Default"> </td><td style="text-align:left;width:5.644cm; " class="Default"> </td><td style="text-align:left;width:19.336cm; " class="Default"> </td><td style="text-align:left;width:11.042cm; " class="Default"> </td><td style="text-align:left;width:13.914cm; " class="Default"> </td><td style="text-align:left;width:11.067cm; " class="Default"> </td><td style="text-align:left;width:19.928cm; " class="Default"> </td><td style="text-align:left;width:19.928cm; " class="Default"> </td><td style="text-align:left;width:24.857cm; " class="Default"> </td><td style="text-align:left;width:2.258cm; " class="ce48"> </td><td style="text-align:left;width:2.258cm; " class="Default"> </td></tr></table></body></html>