<!DOCTYPE html>
  <html lang="en">
    <head>
    <style>
    
        .zebra-striped thead {
  background-color: #9ed8db;
}

.zebra-striped th, td {
  border: 1px solid #ffffff;
}
    
    .zebra-striped tbody tr:nth-child(odd) td, .zebra-striped tbody tr:nth-child(odd) th {
  background-color: #f9f9f9;
}
.zebra-striped tbody tr:hover td, .zebra-striped tbody tr:hover th {
  background-color: #f5f5f5;
}
    </style>
    
    <main>
      <div class='UsrGuideGlobalDiv'>
    <h1 id="usrGuide"><b>User Guide</b></h1>
      <!--Define index with hiperlynks -->
      <!--Poner algo tipo Single Patient Submission y por otro lado Multiple Patient Submission y un fichero de referencia-->
      <ul>
      <li><a href="#ExplanationPostre">What is POSTRE?</a></li>
      <li><a href="#SingleSubm">Single SV Submission</a></li>
      <li><a href="#MultipleSubm">Multiple SV Submission</a></li>
      <li><a href="#ExplorePreviousSVs">Explore Previous SVs</a></li>
      <li><a href="#PathogenicScoreCalculation">Pathogenic score calculation</a></li>
      <li><a href="#DevStages">Cell Types/Tissues considered per Phenotype</a></li>
      <li><a href="#RefGenome">Reference genome - GRCh37/hg19</a></li>
      <li><a href="#PostreInstall">How to install POSTRE?</a></li>
      <li><a href="#Github">POSTRE latest news - Github</a></li>
      <li><a href="#HelpSection">Can we help you?</a></li>
      </ul> 
      
    <h2 id="ExplanationPostre"> <b>What is POSTRE?</b> </h2>
    <div>
    POSTRE is a software developed to <b style='color:#1D3354;'>predict the pathogenic impact of Structural Variants (SVs)</b>. POSTRE aims to identify the genes responsible for the disease together with the pathological mechanism. In parallel, it determines the SV associated pathogenic events <b style='color:#1D3354;'>in space and time</b> through performing cell type/tissue specific predictions. POSTRE is able to handle <b style='color:#1D3354;'>long-range (enhancer driven) and coding pathogenic events</b>. If you want to know more about POSTRE, check our manuscript <a target="_blank" href="https://www.biorxiv.org/content/10.1101/2022.06.20.496902v1">here</a>.
    <br> <br>
     <div>
       <!--Placing here POSTRE infographic -->
       <!--<p align="center"><object data="infografia_user_guide.svg" width="1500"> </object></p>-->
       
       <!--height="1779"-->
       <p align="center"><img src="infografia.png" width="1500"></p>
       
     </div>

    <br>
    <br>
    </div>
    
    <h2 id="SingleSubm"> <b>Single SV Submission</b> </h2>
    <p>Watch POSTRE performance with a real patient in the following YouTube video. Reproduce it in Full Screen and High Quality (1080p) for optimal visualization. For this patient, with BOFS syndrome carrying an inversion (Laugsch et al., 2019), POSTRE successfully predicts the loss of TFAP2A expression in neural crest cells through an enhancer disconnection mechanism.</p>
    <br><br>
          <div class ='videotutorialExplPreviousPat'> 
      <p align="center">
            <iframe width="640" height="360" src="https://www.youtube.com/embed/EmZYSQwfJm0" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
</p>
</div>
    <br><br>
      <p>By default POSTRE presents in the Single SV Submission box the information for the BOFS patient described in the previous video <a target="_blank" href="https://www.sciencedirect.com/science/article/pii/S1934590919301031">Laugsch et al. 2019</a>. To see POSTRE results for this structural variant, click the Submit button in the the <a id="goSinglePage" href="#" role="button" class="btn btn-default action-button">Single SV Submission page</a>.
      </p>
      <br>
      In addition, you can find in the table below the description for a set of 10 patients with pathogenic structural variants and a broad variety of responsible mechanisms. You can run the prediction for those genetic rearrangements with the help of the submission box located below the table.
      <br><br>
      <div>
        <!-- Div, test demo single Submission <table class="demo_table zebra-striped">  <table class="tableUsrGuide">-->
                    <table class="tableUsrGuide"> 
              <thead>
                <tr>
                  <th>SV ID</th>
                  <th>SV Type</th>
                  <th>Mechanism</th>
                  <th>Phenotype</th>
                  <th>Reference</th>
                </tr>
              </thead>
            
              <tbody>
                  <tr>
                  <td>TFAP2A Patient</td>
                  <td>Inversion</td>
                  <td>TFAP2A pathogenic downregulation due to enhancer disconnection</td>
                  <td>Craniofacial </td>
                  <td><a target="_blank" href="https://www.sciencedirect.com/science/article/pii/S1934590919301031">Laugsch et al. 2019</a></td>
                </tr>
                <tr>
                  <td>SOX9 Patient</td>
                  <td>Deletion</td>
                  <td>SOX9 pathogenic downregulation due to enhancer deletion</td>
                  <td>Craniofacial </td>
                  <td><a target="_blank" href="https://www.sciencedirect.com/science/article/pii/S1934590920304471">Long et al. 2020</a></td>
                </tr>
                <tr>
                  <td>KCNJ2 Patient</td>
                  <td>Duplication</td>
                  <td>KCNJ2 pathogenic upregulation due to enhancer adoption through NeoTad formation</td>
                  <td>Limb </td>
                  <td><a target="_blank" href="https://www.nature.com/articles/nature19800">Franke et al. 2016</a></td>
                </tr>
                 <tr>
                  <td>PAX3 Patient</td>
                  <td>Deletion</td>
                  <td>PAX3 pathogenic upregulation due to enhancer adoption through TAD fusion</td>
                  <td>Limb </td>
                  <td><a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/25959774/">Lupiáñez et al. 2015</a></td>
                </tr>
                <tr>
                  <td>IHH Patient</td>
                  <td>Duplication</td>
                  <td>IHH pathogenic upregulation due to enhancer adoption through NeoTad formation</td>
                  <td>Limb </td>
                  <td><a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/25959774/">Lupiáñez et al. 2015</a></td>
                </tr>
                <tr>
                  <td>SATB2 Del Patient</td>
                  <td>Deletion</td>
                  <td>SATB2 deletion responsible of neurodevelopmental alteration</td>
                  <td>Neurodevelopmental </td>
                  <td><a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/21343628/">Balasubramanian et al. 2011</a></td>
                </tr>
                <tr>
                  <td>SATB2 Trunc Patient</td>
                  <td>Deletion</td>
                  <td>SATB2 truncation responsible of neurodevelopmental alteration</td>
                  <td>Neurodevelopmental </td>
                  <td><a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/24363063/">Rainger et al. 2013</a></td>
                </tr>
                <tr>
                  <td>SATB2 LR Patient</td>
                  <td>Translocation</td>
                  <td>SATB2 pathogenic downregulation due to enhancer disconnection</td>
                  <td>Neurodevelopmental </td>
                  <td><a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/24363063/">Rainger et al. 2013</a></td>
                </tr>
                <tr>
                  <td>FOXG1 Patient</td>
                  <td>Translocation</td>
                  <td>FOXG1 pathogenic downregulation due to enhancer disconnection</td>
                  <td>Neurodevelopmental </td>
                  <td><a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/29289958/">Mehrjouy et al. 2017</a></td>
                </tr>
                <tr>
                  <td>TBX3&5 Patient</td>
                  <td>Deletion</td>
                  <td>Cardiovascular deffect triggered by the additive effect of the deletion of TBX3 and TBX5</td>
                  <td>Cardiovascular </td>
                  <td><a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/23713051/">Alby et al. 2013</a></td>
                </tr>
              </tbody>
            </table>
            
<div class="titleSubmissionBoxUserGuideSinglePatient">
<h3><center><b style="color:#467599;">Run prediction with the structural variant of interest</b></center></h3>
</div>

<!--Formatting submission box for UserGuide Selected Patient -->
<div class=panelSubm_UsrGuide_SinglePatient>

<div class="usrGuideSelectedPatient well">

    <div class="form-group shiny-input-container">
      <label class="control-label" id="SV of interest" for="SV of interest">Choose structural variant</label>
      <div>
<select id="singlePatientId">
<option value="TFAP2A_patient" selected="selected">TFAP2A Patient</option>
<option value="SOX9_patient_F1">SOX9 Patient</option>
<option value="KCNJ2_patient">KCNJ2 Patient</option>
<option value="PAX3_patient">PAX3 Patient</option>
<option value="IHH_patient">IHH Patient</option>
<option value="SATB2_Del_pat4_PMID:21343628">SATB2 Del Patient</option>
<option value="SATB2_Trunc_case2Intrag_PMID:24363063">SATB2 Trunc Patient</option>
<option value="SATB2_LR_case1_PMID:24363063">SATB2 LR Patient</option>
<option value="FOXG1_P2_CLINVAR_SCV000576474_PMID:29289958">FOXG1 Patient</option>
<option value="TBX3&5_case3_PMID:23713051">TBX3&5 Patient</option>
</select>
 <!--          <script type="application/json" data-for="typeSV" data-nonempty="">{"plugins":["selectize-plugin-a11y"]}</script> -->
      </div>
    </div>
</div> 
   
   <!-- SUBMISSION BOX -->
   <div class="usrGuideSinglePatient_SubmissionButton">
    <button id="click_SingleSubmissionUserGuide" type="button" class="btn btn-default action-button" style="color: #fff; background-color: #1D3354; border-color: #467599;">
  <i class="fa fa-paper-plane" role="presentation" aria-label="paper-plane icon"></i>
  Submit
</button>
   </div>         
</div> <!--Closing panelSubm_UsrGuide_SinglePatient -->     
        
      </div>

    <br>
    <br>
    
      <h2 id="MultipleSubm"> <b>Multiple SV Submission</b> </h2>
      The Multiple SV Submission box allows the sequential analysis of multiple structural variants (results can be downloaded as txt tables). The SVs may come from just one or multiple patients. The SVs information has to be uploaded in a specific format. The file format consists on 1 line and 7 columns per structural variant. Each structural variant must contain only two breakpoints. The information associated with each column is provided below:
      <br><br>
      <i>Note: For the case of structural variants happening strictly in one chromosome (deletions, inversions, duplications) the breakpoint 1 is the one associated with a smaller genomic coordinate, and the breakpoint 2 the one associated with a larger genomic coordinate. For translocations, it does not matter.</i>
      <br><br>
      <ul>
      <li>Column 1: Chromosome for the breakpoint 1  </li>
      <li>Column 2: Genomic coordinates for the breakpoint 1. When not base pair resolution, provide a comma separated range, e.g. 85092268,85092269.  </li>
      <li>Column 3: Chromosome for the breakpoint 2</li>
      <li>Column 4: Genomic coordinates for the breakpoint 2. When not base pair resolution, provide a comma separated range, e.g. 85092268,85092269.</li>
      <li>Column 5: Structural Variant Type. Current options: Inversion, Translocation, Deletion or Duplication.</li>
      <li>Column 6: Comma separated list of phenotypes associated with the structural variant. Current options are: head_neck, limbs, neurodevelopmental or cardiovascular. For instance: head_neck,neurodevelopmental,cardiovascular.  </li>
      <li>Column 7: Structural variant unique identifier e.g. (Patient1_SV3)</li>
      </ul> 
      
    The data must be stored in a plain text file with column values separated by tabulations
    
    <br><br>
    An example file can be downloaded <a href="ExampleMultipleSubmission.tsv" target="_blank">here</a>.  Additional test files can also be downloaded from the <a href="https://github.com/vicsanga/Postre/tree/main/testFiles" target="_blank">GitHub-testFiles</a> folder (to download them: (1) click on the file of interest, (2) right-click the "Raw" button at the top of the file, (3) select Save Link As…, (4) choose the location on your computer where you want to save the file, and (5) select Save).
    
    <br><br>
    Upon the analysis of multiple SVs two main tables are provided. The first one (Results per SV and phenotype) is a table with a pathogenic prediction for each of the SVs and associated phenotypes analyzed. The second one (Results per gene and phenotype) is an aggregation of the pathogenic predictions per gene, phenotype and pathogenic mechanism (coding, long-range). Perform a Multiple SV Submission with one of the test files and check "How to navigate through this page?" section for more details.
    <br><br>
A tutorial video showing how to perform and interpret results for a Multiple SV Submission analysis is provided in the following video.
    <br><br>
          <div class ='videotutorialExplPreviousPat'> 
      <p align="center">
            <iframe width="640" height="360" src="https://www.youtube.com/embed/UBYoa79hJko" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
</p>
</div>


    <br>
    <br>
    
    <h2 id="ExplorePreviousSVs"> <b>Explore Previous SVs</b> </h2>
    The Explore Previous SVs page presents the results for the analysis of SVs from different sources/databases. Current results come from the analysis of SVs from <a href="https://pubmed.ncbi.nlm.nih.gov/27841880/" target="_blank">(Redin et al., 2017)</a> and <a href="https://pubmed.ncbi.nlm.nih.gov/31801603/" target="_blank">(Middelkamp et al., 2019)</a>. Only results for SVs where pathogenicity has been predicted are shown.
    
    
    <h2 id="PathogenicScoreCalculation"> <b>Pathogenic score calculation</b> </h2>
    <div>
      To get a simplified explanation of POSTRE pathogenic score calculation, please visit the following <a href="PathogenicScoreCalculation.html" target="_blank"> file </a>.
    <br>
    </div>
    
    
    <h2 id="DevStages"> <b>Cell Types/Tissues considered per Phenotype</b> </h2>
      If you want to know more about the different cell types/tissues considered for each phenotype, check the 
    <a href="Supplementary_Data_1.html" target="_blank"> Cell types/tissues per phenotype File </a>. 
    <br>
    <br>
    Are you missing some data? Do you want us to add some data which might be profitable for your research? Tell us about it through postre.radaiglesiaslab@gmail.com
    <br>
    <br>
    
    <h2 id="RefGenome"> <b>Reference genome - GRCh37/hg19</b> </h2>
      The reference genome currently handled by POSTRE is GRCh37/hg19. Accordingly, all genomic coordinates provided must be in this reference genome. Please, visit any of the following websites to convert your coordinates to hg19 if you need it:
      <a href= "http://genome.ucsc.edu/cgi-bin/hgLiftOver" target="_blank">UCSC LiftOver</a>, 
      <a href= "https://www.ensembl.org/Homo_sapiens/Tools/AssemblyConverter?db=core" target="_blank">ENSEMBL Assembly Converter</a>, 
      <a href= "https://liftover.broadinstitute.org/" target="_blank">Broad Institute LiftOver</a>
    <br>
    <br>
    
    <h2 id="PostreInstall"> <b>How to install POSTRE?</b> </h2>
    The details for POSTRE installation are provided on <a href="https://github.com/vicsanga/Postre" target="_blank">POSTRE Github page</a>.
    <br>
    <br>
    
    <h2 id="Github"> <b>POSTRE latest news - Github</b> </h2>
    Any changes on the tool will be posted on <a href="https://github.com/vicsanga/Postre" target="_blank">POSTRE Github page</a>.
    <br>
    <br>
    
    <h2 id="HelpSection"> <b>Can we help you?</b> </h2>
      If you are experiencing some trouble, or you just want to contact us for any reason, please send a mail to postre.radaiglesiaslab@gmail.com.
    <br>
    <br>
    

</div> <!--Closing: UsrGuideGlobalDiv-->
</main>
      
</html>